### PR TITLE
Collatz conjecture exercise

### PR DESCRIPTION
**Pull request by pclausen**
Adding Collatz conjecture exercise

### CODE REVIEW

File: bin/create_fortran_test.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def create_test(test_name, json_name):
     j = None
     with open(json_name) as f:
         j = json.load(f)
-    exercise_version = j['version']
+    exercise_version = j['version'] if 'version' in j else None
```
Comment (SaschaMann) [inline bin/create_fortran_test.py:122]: The JSON files no longer have versions (see https://github.com/exercism/problem-specifications/issues/1674), so this will always be `None`.

**Review by SaschaMann — COMMENTED** [review]
I think this is missing a README file. Looks good otherwise.

File: bin/create_fortran_test.py (hunk 1 above)
Comment (pclausen) [reply bin/create_fortran_test.py]: Thanks, removed version

File: exercises/acronym/.meta/tests.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[canonical-tests]
```
Comment (SaschaMann) [inline exercises/acronym/.meta/tests.toml:1]: This and the tests.toml file below seem unrelated to the PR. Merging probably won't cause any issues, just pointing it out in case they were committed by accident.

File: exercises/collatz-conjecture/collatz_conjecture_test.f90 [hunk 3]
```diff
@@ -0,0 +1,35 @@
+
+! This test was created from E:\nosave\code\gitrepos\pclausen\problem-specifications\exercises\collatz-conjecture\canonical-data.json
```
Comment (SaschaMann) [inline exercises/collatz-conjecture/collatz_conjecture_test.f90:2]: You might want to strip the first part of the path from this comment as it's specific to your computer.

File: exercises/collatz-conjecture/collatz_conjecture_test.f90 [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+! version: None
+!
+
+program collatz_conjecture_test_main
+  use TesterMain
+  use collatz_conjecture
+  implicit none
+
+  ! Test 1: zero steps for one
+  ! UUID 1: 540a3d51-e7a6-47a5-92a3-4ad1838f0bfd
```
Comment (SaschaMann) [inline exercises/collatz-conjecture/collatz_conjecture_test.f90:12]: The test UUIDs are mostly meant for track tooling. It's best to not expose them to the students and instead only leave the comments.

**Review by SaschaMann — APPROVED** [review]
Left some comments but feel free to merge it as is, too.
